gpii/node_modules/lifecycleManager/src/LifecycleManager.js

```diff
@@ -568,13 +587,79 @@ var gpii = fluid.registerNamespace("gpii");
         fluid.set(fullPayload, ["matchMakerOutput", "inferredConfiguration"], newInferred);
         that.addLifecycleInstructionsToPayload(fullPayload);
 
-        session.applier.change(["activeConfiguration"], fullPayload.activeConfiguration);
+        session.applier.change("", {
+            activeConfiguration: fullPayload.activeConfiguration,
+            currentPreferences: gpii.lifecycleManager.getContextedPreferences(fullPayload.preferences, fullPayload.activeContextName)
+        });
 
         that.update(fullPayload).then(prefsAppliedLocalEvent.fire);
     };
 
-    gpii.lifecycleManager.logSnapshotUpdate = function (lifecycleManager, session, originalSettings) {
-        fluid.log("Settings for session " + session.id + " created at " + session.createTime + " updated to ", originalSettings);
+    /**
+     * Read corresponding setting values of the requested preferences. The read result is applied as
+     * a model change to userSession.model.currentPreferences.
+     * @param {Component} that - An instance of gpii.lifecycleManager.
+     * @param {Object} preference - The preference to read settings for. An example is:
+     * "http://registry\\.gpii\\.net/common/magnification": {
+     *     value: 3
+     * }
+     * @param {Event} preferenceReadSuccessEvent - The event fired when the read succeeds.
+     * @param {Event} preferenceReadFailEvent - The event fired when the read fails.
+     */
+    gpii.lifecycleManager.readPreference = function (that, preference, preferenceReadSuccessEvent, preferenceReadFailEvent) {
+        fluid.log("Lifecycle Manager: Received preference to read ", preference);
+        var session = that.getSession();
+
+        // Mock values for running preference thru matchMaker
+        var contextualPrefs = {};
```

Comment:
As below, supply better names for this overall prefs document, and avoid any fresh reference to "context" in new code being contributed. activePrefsSetName is better